src/main/java/gameframework/motion/blocking/MoveBlockerRulesApplierDefaultImpl.java

```diff
@@ -53,14 +57,37 @@ public boolean moveValidationProcessing(GameMovable movable,
 		return true;
 	}
 
-	private void moveBlockerRuleApply(GameMovable movable, MoveBlocker blocker)
+	/**
+	 * Applies the moveBlockerRule by calling the correct method depending of the
+	 * parameters type.
+	 * @param movable
+	 * @param blocker
+	 * @throws Exception
+	 */
+	protected void moveBlockerRuleApply(GameMovable movable, MoveBlocker blocker)
 			throws Exception {
-		Method m = null;
-		m = (getClass()).getMethod("moveBlockerRule", movable.getClass(),
+		Method m = (getClass()).getMethod("moveBlockerRule", movable.getClass(),
 				blocker.getClass());
 		m.invoke(this, movable, blocker);
 	}
 
+
+	/**
+	 * This method checks if the movable is allowed to move by the blocker.
+	 * If not, this method will throw an IllegalMoveException. Otherwise, it
+	 * does nothing.
+	 *
+	 * It is a default behaviour, that you can expend by adding other moveBlockerRule
+	 * methods to have different behaviours depending on the type of the parameters.
+	 * @param movable the movable to check
+	 * @param blocker the blocker that might block the movable
+	 * @throws IllegalMoveException
+	 */
+	protected void moveBlockerRule(GameMovable movable, MoveBlocker blocker)
```

Comment:
I did it because I found it difficult to understand what was going on in this class. That way, I showed a simple default behaviour as an example for anyone trying to use this class. By adding a method, I was also able to document it and to add more clarity once again.